src/System.Text.Json/src/System/Text/Json/Serialization/MemberAccessor.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json.Serialization
+{
+    internal static class MemberAccessor
+    {
+        private delegate TProperty GetProperty<TClass, TProperty>(TClass obj);
+        private delegate TProperty GetPropertyByRef<TClass, TProperty>(ref TClass obj);
+
+        private delegate void SetProperty<TClass, TProperty>(TClass obj, TProperty value);
+        private delegate void SetPropertyByRef<TClass, TProperty>(ref TClass obj, TProperty value);
+
+        private delegate Func<object, TProperty> GetPropertyByRefFactory<TClass, TProperty>(GetPropertyByRef<TClass, TProperty> set);
+        private delegate Action<object, TProperty> SetPropertyByRefFactory<TClass, TProperty>(SetPropertyByRef<TClass, TProperty> set);
+
+        private static readonly MethodInfo s_createStructPropertyGetterMethod = new GetPropertyByRefFactory<int, int>(CreateStructPropertyGetter)
+            .Method.GetGenericMethodDefinition();
+
+        private static readonly MethodInfo s_createStructPropertySetterMethod = new SetPropertyByRefFactory<int, int>(CreateStructPropertySetter)
+            .Method.GetGenericMethodDefinition();
+
+        internal static Func<object, TProperty> CreatePropertyGetter<TClass, TProperty>(PropertyInfo propertyInfo)
+        {
+            MethodInfo getMethodInfo = propertyInfo.GetGetMethod();
+
+            if (typeof(TClass).IsValueType)
+            {
+                var factory = CreateDelegate<GetPropertyByRefFactory<TClass, TProperty>>(s_createStructPropertyGetterMethod.MakeGenericMethod(typeof(TClass), typeof(TProperty)));
+                var propertyGetter = CreateDelegate<GetPropertyByRef<TClass, TProperty>>(getMethodInfo);
```

Comment:
nit: Same here/elsewhere (use of var).

Reply:
Visual Studio suggests to use the concrete type here, but it isn't clever enough to determine that it's already specified as a parameter of `CreateDelegate`.

Reply:
> but it isn't clever enough 

As a code reviewer, neither am I (or rather, I prefer the convenience of the type being explicitly stated in-place).